tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/bala/object/ObjectInBalaTest.java

```diff
@@ -546,6 +547,27 @@ public void testObjectReferingTypeFromBalaNegative() {
                 "no implementation found for the method 'toString' of class 'FrameImpl'", 53, 1);
     }
 
+    @Test (description = "Negative test to verify object qualifiers load properly from BIR")
+    public void testDistinctIsolatedObjectsNegative() {
+        CompileResult result = BCompileUtil.compile("test-src/bala/test_bala/object/test_bir_negative.bal");
+        int i = 0;
+        BAssertUtil.validateError(result, i++, 
+                "incompatible types: 'bir/objs:0.1.0:Bar' will not be matched to 'bir/objs:0.1.0:Foo'", 20, 9);
+        BAssertUtil.validateError(result, i++, 
+                "incompatible types: 'bir/objs:0.1.0:Foo' will not be matched to 'bir/objs:0.1.0:Bar'", 24, 9);
+        BAssertUtil.validateError(result, i++, 
+                "incompatible types: 'bir/objs:0.1.0:Xyz' will not be matched to 'bir/objs:0.1.0:Qux'", 28, 9);
+        BAssertUtil.validateError(result, i, 
+                "incompatible types: 'bir/objs:0.1.0:Qux' will not be matched to 'bir/objs:0.1.0:Xyz'", 32, 9);
+        Assert.assertEquals(result.getErrorCount(), 4);
+    }
+
+    @Test
+    public void testObjectTypeAssignability() {
```

Comment:
Can we be more descriptive here, mentioning flags/qualifiers?

Reply:
Added in [#050dc4](https://github.com/ballerina-platform/ballerina-lang/pull/40901/commits/050dc4b510554b9df7e59929bf5265976524497b)